lms/djangoapps/bookmarks/models.py

```diff
@@ -60,12 +123,63 @@ def get_path(block):
         Returns:
             list of dicts of the form {'usage_id': <usage_id>, 'display_name': <display_name>}.
         """
-        parent = block.get_parent()
-        parents_data = []
+        try:
+            path = path_to_location(modulestore(), usage_key, full_path=True)
+        except ItemNotFoundError:
+            log.error(u'Block with usage_key: %s not found.', usage_key)
+            return []
+        except NoPathToItem:
+            log.error(u'No path to block with usage_key: %s.', usage_key)
+            return []
+
+        path_data = []
+        for ancestor_usage_key in path:
+            if ancestor_usage_key != usage_key and ancestor_usage_key.block_type != 'course':
+                try:
+                    block = modulestore().get_item(ancestor_usage_key)
```

Comment:
@cpennington I am unable to `get_items()` to accept a list of UsageKeys like this. And the docstring doesn't say anything about this usage either: https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/__init__.py#L761